tests/api_fastapi/core_api/routes/public/test_assets.py

```diff
@@ -310,6 +310,20 @@ def test_should_respect_page_size_limit_default(self, test_client):
         assert response.status_code == 200
         assert len(response.json()["assets"]) == 100
 
+    @pytest.mark.parametrize(
+        "query_params, expected_error_param",
+        [
+            ({"limit": 1, "offset": -1}, "Offset"),
+            ({"limit": -1, "offset": 1}, "Limit"),
+            ({"limit": -1, "offset": -1}, "Limit"),
+            ({"offset": -1, "limit": -1}, "Limit"),
```

Comment:
The order of parameters defined on the endpoint decides which error will be emitted. I'm not sure if we can collect all validation errors and return them together.